### PR TITLE
API Contract to retrieve individual Progress Document and trackedProgresses Collection

### PR DESCRIPTION
**Pull request by heyrandhir**
This PR adds API contract documentation for the new endpoint `/progresses/:type/:typeId/date/:date`. The API contract provides clear and detailed information about how to retrieve progress documents based on a specific user ID or task ID and a specified date.

This PR also adds API contract documentation for the new endpoint `/trackedProgresses`.

The Parent ticket can be found [here](https://github.com/Real-Dev-Squad/website-backend/issues/1076)

### CODE REVIEW

**Review by sahsisunny — COMMENTED** [review]
LGTM

**Review by sahsisunny — APPROVED** [review]

**Review by Pratiyushkumar — COMMENTED** [review]
LGTM, waiting for other approvals

**Review by Pratiyushkumar — DISMISSED** [review]
LGTM

File: progress/README.md [hunk 1]
```diff
@@ -2,11 +2,12 @@
...
-| [GET /progresses/range](#get-progressesrange) | Retrieves the progress records for a particular user or task within the specified date range. |
+|                     Route                     |                                           Description                                           |
+| :-------------------------------------------: | :---------------------------------------------------------------------------------------------: |
+|     [POST /progresses](#post-progresses)      |                                  Creates a new progress entry.                                  |
+|      [GET /progresses](#get-progresses)       |                  Retrieves progress entries based on the provided parameters.                   |
+| [GET /progresses/range](#get-progressesrange) |  Retrieves the progress records for a particular user or task within the specified date range.  |
+| [GET /progresses/:type/:typeId/date/:date](#get-progressestypetypeiddatedate)  | Retrieves the progress documents based on a specific user ID or task ID for the specified date. |
 
 ## POST /progresses
```
Comment (Pratiyushkumar) [inline progress/README.md:12]: what if we only provide type and typeId and don't provide date, then what will be response

Comment (heyrandhir) [reply progress/README.md:12]: No you will not get a response as this path would not match and you will get an error not found

**Review by Pratiyushkumar — DISMISSED** [review]
LGTM

**Review by shubham-y — APPROVED** [review]

**Review by sahsisunny — APPROVED** [review]

**Pull request merged**